src/javascript/app/pages/user/account/change_password.js

```diff
@@ -1,45 +1,259 @@
-const BinaryPjax   = require('../../../base/binary_pjax');
-const Client       = require('../../../base/client');
-const BinarySocket = require('../../../base/socket');
-const FormManager  = require('../../../common/form_manager');
-const localize     = require('../../../../_common/localize').localize;
-const State        = require('../../../../_common/storage').State;
+const Client         = require('../../../base/client');
+const BinarySocket   = require('../../../base/socket');
+const FormManager    = require('../../../common/form_manager');
+const Dialog         = require('../../../common/attach_dom/dialog');
+const toTitleCase    = require('../../../../_common/string_util').toTitleCase;
+const localize       = require('../../../../_common/localize').localize;
+const State          = require('../../../../_common/storage').State;
+const getElementById = require('../../../../_common/common_functions').getElementById;
+const Url            = require('../../../../_common/url');
+const Password       = require('../../../../_common/check_password');
 
 const ChangePassword = (() => {
-    const form_id = '#frm_change_password';
+    let $change_password_loading,
+        $change_password_container,
+        binary_form_id,
+        trading_form_id,
+        social_unlink_btn_id,
+        $binary_password_container,
+        $social_signup_container,
+        $trading_password_container,
+        $set_trading_password_container,
+        social_signup_identifier,
+        $msg_success_container,
+        $msg_success,
+        $msg_success_trading_container,
+        $msg_success_trading,
+        forgot_binary_pw_btn_id,
+        forgot_trading_pw_btn_id;
+
+    const social_signups = {
+        'Google': {
+            icon: 'ic-linked-google',
+            name: 'Google',
+        },
+        'Facebook': {
+            icon: 'ic-linked-facebook',
+            name: 'Facebook',
+        },
+        'Apple': {
+            icon: 'ic-linked-apple',
+            name: 'Apple',
+        },
+    };
+
+    const shouldSetTradingPassword = () => {
+        const { status } = State.getResponse('get_account_status');
+        if (Array.isArray(status)) {
+            return status.includes('trading_password_required');
+        }
+        throw new Error('get_account_status is undefined');
+    };
+
+    const hasSocialSignup = () => {
+        const { social_identity_provider, status } = State.getResponse('get_account_status');
+        if (Array.isArray(status)) {
+            social_signup_identifier = toTitleCase(social_identity_provider);
+            return status.includes('social_signup');
+        }
+        throw new Error('get_account_status is undefined');
```

Comment:
same here

Reply:
it's best to wait for get_account_status in the onload function before doing anything else if we need it there.